### PR TITLE
Update Vulkan and related libraries to 1.3.250.0

### PR DESCRIPTION
**Pull request by DeeJayLSP**
Updates glslang, spirv-reflect, volk and vulkan to 1.3.250.0.

Some patches might need some changes (one was implemented, so I deleted).

### CODE REVIEW

File: thirdparty/glslang/SPIRV/doc.cpp [hunk 1]
```diff
@@ -53,6 +53,7 @@ namespace spv {
         #include "GLSL.ext.EXT.h"
```
Comment (akien-mga) [inline thirdparty/glslang/SPIRV/doc.cpp:53]: Wrong permission change.

File: thirdparty/glslang/glslang/Include/BaseTypes.h [hunk 2]
```diff
@@ -65,6 +65,7 @@ enum TBasicType {
     EbtAccStruct,
```
Comment (akien-mga) [inline thirdparty/glslang/glslang/Include/BaseTypes.h:65]: Permission change.

File: thirdparty/glslang/glslang/MachineIndependent/Initialize.cpp [hunk 3]
```diff
@@ -147,10 +147,6 @@ EProfile EDesktopProfile = static_cast<EProfile>(ENoProfile | ECoreProfile | ECo
 #ifdef GLSLANG_WEB
```
Comment (akien-mga) [inline thirdparty/glslang/glslang/MachineIndependent/Initialize.cpp:147]: Permission change.

File: thirdparty/glslang/glslang/MachineIndependent/Versions.h [hunk 4]
```diff
@@ -163,6 +163,7 @@ const char* const E_GL_ARB_texture_query_lod            = "GL_ARB_texture_query_
 const char* const E_GL_ARB_vertex_attrib_64bit          = "GL_ARB_vertex_attrib_64bit";
```
Comment (akien-mga) [inline thirdparty/glslang/glslang/MachineIndependent/Versions.h:163]: Permission change.

**Review by akien-mga — COMMENTED** [review]
Looks great, thanks for working on this!

Some further copyright statements to update to match what upstream projects did:
```diff
diff --git a/COPYRIGHT.txt b/COPYRIGHT.txt
index 446f833eb9..497b4a535c 100644
--- a/COPYRIGHT.txt
+++ b/COPYRIGHT.txt
@@ -425,7 +425,7 @@ License: Apache-2.0
 
 Files: ./thirdparty/spirv-reflect/
 Comment: SPIRV-Reflect
-Copyright: 2017-2018, Google Inc.
+Copyright: 2017-2022, Google Inc.
 License: Apache-2.0
 
 Files: ./thirdparty/squish/
@@ -452,15 +452,15 @@ License: BSD-3-clause
 
 Files: ./thirdparty/volk/
 Comment: volk
-Copyright: 2018-2019, Arseny Kapoulkine
+Copyright: 2018-2023, Arseny Kapoulkine
 License: Expat
 
 Files: ./thirdparty/vulkan/
 Comment: Vulkan Headers
-Copyright: 2014-2021, The Khronos Group Inc.
-  2014-2021, Valve Corporation
-  2014-2021, LunarG, Inc.
-  2015-2021, Google Inc.
+Copyright: 2015-2023, The Khronos Group Inc.
+  2015-2023, Valve Corporation
+  2015-2023, LunarG, Inc.
+  2015-2023, Google Inc.
 License: Apache-2.0
 
 Files: ./thirdparty/vulkan/vk_mem_alloc.h
diff --git a/thirdparty/volk/LICENSE.md b/thirdparty/volk/LICENSE.md
index e5e847ad25..7315252211 100644
--- a/thirdparty/volk/LICENSE.md
+++ b/thirdparty/volk/LICENSE.md
@@ -1,4 +1,4 @@
-Copyright (c) 2018-2022 Arseny Kapoulkine
+Copyright (c) 2018-2023 Arseny Kapoulkine
 
 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
```

**Comment by DeeJayLSP** [comment]
All fixed I guess

**Review by akien-mga — APPROVED** [review]
Looks good to me. We'll merge after beta 1 is released as that build is imminent, and Vulkan updates can sometimes cause regressions / expose platform specific issues.

**Comment by DeeJayLSP** [comment]
@akien-mga Running any project still says the Vulkan version is 1.3.236. Is this right or something I forgot?

**Comment by akien-mga** [comment]
That's probably the Vulkan version installed on your OS which is reported.

Updating our version gives us access to more APIs but at the end it's the installed driver version which matters.

**Comment by DeeJayLSP** [comment]
Weird because my OS's Vulkan is actually older than that

**Comment by akien-mga** [comment]
It should report the `api_version` from the `icd` file it's using. For me on Linux with Mesa 23.1.2:
```
$ grep api_version /usr/share/vulkan/icd.d/radeon_icd.x86_64.json 
        "api_version": "1.3.246",
```

So that seems to be the version Mesa 23.1.2 compiles for, despite my distro Vulkan-Loader being a bit older too (1.3.231.1).

**Pull request merged**